src/app/main.rs

```diff
@@ -290,22 +299,34 @@ where
                     .push(normalize_uri::layer())
                     .push(orig_proto_upgrade::layer())
                     .push(tap::layer(tap_next_id.clone(), taps.clone()))
-                    .push(metrics::layer::<_, classify::Response>(http_metrics))
-                    .push(classify::layer())
+                    .push(metrics::layer::<_, classify::Response>(endpoint_http_metrics))
                     .push(svc::watch::layer(tls_client_config))
                     .push(buffer::layer());
 
-                let dst_router_stack = endpoint_stack
+                let profiles_client = ProfilesClient::new(
+                    controller,
+                    Duration::from_secs(3),
+                    KubernetesNormalizer::new(config.namespaces.pod.clone()),
+                );
+
+                let dst_route_stack = endpoint_stack
                     .push(resolve::layer(Resolve::new(resolver)))
                     .push(balance::layer())
                     .push(buffer::layer())
+                    .push(profiles::router::layer(
+                        profiles_client,
+                        svc::stack::phantom_data::layer()
```

Comment:
Just curious --- is the `phantom_data` layer here needed because of type inference issues, or is it just used to push the metrics and classification layers to something? I'd like a better understanding of when/where that layer needs to be used...

Reply:
me too. afaict this is needed for inference.